src/os.c

```diff
@@ -432,7 +432,7 @@ int shell_cmd(const char *substs[][2], const char *args_str, int silent)
         }
         execvp(args[0], args);
         _exit(1);
-    } else if (waitpid(child, &exit_status, 0) == (pid_t) -1 || !WIFEXITED(exit_status)) {
+    } else if (waitpid(child, &exit_status, WNOHANG|WUNTRACED) == (pid_t) -1 || !WIFEXITED(exit_status)) {
```

Comment:
`WNOHANG` shouldn't hurt, but if a child has been stopped, we want to wait for it. I'm not sure what `WUNTRACED` is a good idea here. Maybe `WCONTINUED`, but that's a Linux-only thing.